### PR TITLE
Feature/add Codestral

### PR DESCRIPTION
**Pull request by bhodrolok**
### Issue
#48 

### Description
Adds support for Mistral AI's new genAI model, Codestral.

### Testing
Manually by staging (adding to the index) multiple files, running the tool, and checking if there was generated commit messages from `codestral.mistral.ai`. 

Also tested it on other projects by [publishing it](https://github.com/bhodrolok/aicommit2-codestral/tree/npm/main) and using it as outlined in the [contributing guide](https://github.com/tak-bro/aicommit2/blob/main/CONTRIBUTING.md#using--testing-your-changes).

Development was done on a machine Windows and looks like some of the automatic test cases would have been skipped because of this (a NodeJS bug apparently).

### Additional context
I think the current approach here is a more _refactored_ one, in the sense that the preexisting `MistralService` class is (re)used instead of creating a new `CodestralService`. Went with this because the only tangible difference was in the domain where the requests were sent to and the fact that this domain also didn't have a [`List All Models`](https://docs.mistral.ai/api/#operation/listModels) endpoint like `api.mistral.ai`.

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/services/ai/mistral.service.ts [hunk 1]
```diff
@@ -44,18 +44,24 @@ export interface CreateChatCompletionsResponse {
 }
 
 export class MistralService extends AIService {
-    private host = `https://api.mistral.ai`;
```
Comment (tak-bro) [inline src/services/ai/mistral.service.ts:47]: Could you please use Camelcase for variables?

File: src/utils/config.ts [hunk 2]
```diff
@@ -210,9 +210,9 @@ const configParsers = {
     },
     GEMINI_MODEL(model?: string) {
         if (!model || model.length === 0) {
-            return 'gemini-1.5-pro-latest';
+            return 'gemini-1.5-flash-latest';
```
Comment (tak-bro) [inline src/utils/config.ts:213]: Please update this code as [pull request](https://github.com/tak-bro/aicommit2/pull/47)

**Review by tak-bro — COMMENTED** [review]
Thank you for pull request.
Could you please check my review?

File: src/services/ai/mistral.service.ts (hunk 1 above)
Comment (bhodrolok) [reply src/services/ai/mistral.service.ts:47]: Updated some snake_case names to camelCase.

File: src/utils/config.ts (hunk 2 above)
Comment (bhodrolok) [reply src/utils/config.ts]: Updated!

**Comment by bhodrolok** [comment]
Thanks for the review! Updated code as per your suggestions. There is a small issue where the same display provider is shown to the user even if the requests themselves come from Codestral and Mistral's other models separately.
image


I suspect it is because of the switch case handling in `src/managers/ai-request.manager.ts`, specifically [here](https://github.com/tak-bro/aicommit2/pull/49/files?diff=split&w=0#diff-7e6042904fc2ba151d29dd6f45a84c9651825dc49ce004a2a527c830b56f8d5dR44-R46).

[Here] is the statement that controls the provider to be displayed.

**Comment by tak-bro** [comment]
Thank you very much. I'm going to look at the issue you mentioned.

**Comment by tak-bro** [comment]
@bhodrolok Considering the usability of Mistral along with the issues you mentioned, I thought it would be better to divide the `CordestralService` and `MistralService`. Please understand.

**Pull request merged**